File: CSW/JavaScript/220908_최소원_배열복원하기.js
var fs = require('fs');
var input = fs.readFileSync(`/dev/stdin`).toString().trim().split('\n');

const [AR, AC, BR, BC] = input[0].split(' ').map(e=>parseInt(e));
const arrB = [];

for(let i=1; i<=AR+BR; i++){
    let col = input[i].split(' ').map(e=>parseInt(e));
    arrB.push(col);
}

for(let i=BR; i<AR+BR; i++){
    for(let j=BC; j<AC+BC; j++){
        arrB[i][j] -= arrB[i-BR][j-BC];
    }
}


for(let i=0; i<AR; i++){
    let answer = arrB[i].slice(0, AC).join(' ');
    console.log(answer);
}

File: CSW/C++/220908_최소원_배열복원하기.cpp
#include <iostream>
using namespace std;

int AR, AC;
int BR, BC;
int arrB[600][600];

int main() {
	
	cin >> AR >> AC >> BR >> BC;

	for (int i = 0; i < AR + BR; i++) {
		for (int j = 0; j < AC + BC; j++) {
			cin >> arrB[i][j];
		}
	}

	for (int i = BR; i < AR + BR; i++) {
		for (int j = BC; j < AC + BC; j++) {
			arrB[i][j] -= arrB[i - BR][j - BC];
		}
	}

	for (int i = 0; i < AR; i++) {
		for (int j = 0; j < AC; j++) {
			cout << arrB[i][j] <<" ";
		}
		cout << "\n";
	}


	return 0;
}